alma/controllers/hook/DisplayPaymentReturnHookController.php

```diff
@@ -45,7 +45,7 @@ public function run($params)
 
         $payment = null;
         $order = array_key_exists('objOrder', $params) ? $params['objOrder'] : $params['order'];
-        $orderPayment = $this->getOrderPaymentOrFail($order);
+        $orderPayment = $this->getOrderPaymentOrFail($order, false);
```

Comment:
you should use another method instead of changing the method behavior by adding boolean parameter.

Your method should have a Single responsibility (S of SOLID).

You probably have to develop a `getOrderPayment` that's will throws `RenderPaymentException` and use it when you don't want an ajax `OrFail`